lib/puppet/provider/templates/webadministration/logproperties.ps1.erb

```diff
@@ -1,9 +1,44 @@
+$iis_version = '<%= Facter.value(:iis_version) %>'
+
+$W3CValue = 'W3C'
+$MaxSizeValue = 'MaxSize'
+
+# Convert the string representation of an Enum value into an Int32 safely
+Function ConvertStringTo-EnumInt32($EnumType, $Value, $ValueName) {
+  $EnumValue = ([Enum]::GetNames($EnumType)) | ForEach-Object {
+  if ($_.ToUpper() -eq $Value.ToUpper()) { return [Int32]($EnumType::$_) }
+  }
+  if ($EnumValue -eq $null) {
+    Throw "$Value is not a valid value for $ValueName"
+  } else {
+    Return $EnumValue
+  }
+}
+# Microsoft.Web.Administration namespace - https://msdn.microsoft.com/en-us/library/microsoft.web.administration(v=vs.90).aspx
+# Log Format - https://msdn.microsoft.com/en-us/library/microsoft.web.administration.logformat(v=vs.90).aspx
+# LogExtFileFlags - https://msdn.microsoft.com/en-us/library/microsoft.web.administration.logextfileflags(v=vs.90).aspx
+# LogPeriod - https://msdn.microsoft.com/en-us/library/microsoft.web.administration.loggingrolloverperiod(v=vs.90).aspx
+
+if ($iis_version -eq '7.5') {
+  # In IIS 7.5 the Enums are saved using the Int32 value, whereas in later versions, the text value can be used
+  # Load the WebAdministration DLL so we can convert string values into their Int32 enum equivalent
+  [System.Reflection.Assembly]::LoadWithPartialName("Microsoft.Web.Administration") | Out-Null
```

Comment:
I don't think it's appropriate to add the DLL load in there.  That function is entirely generic and can work on any Enum.  Where the DLL load is specific to IIS 7.5 only.